src/tools/tool.ts

```diff
@@ -27,6 +27,22 @@ export abstract class ToolBase {
 
     protected abstract argsShape: ZodRawShape;
 
+    protected get annotations(): ToolAnnotations {
+        const annotations: ToolAnnotations = {
+            title: this.name,
+            description: this.description,
+        };
+
+        if (this.operationType === "read" || this.operationType === "metadata") {
+            annotations.readOnlyHint = true;
+            annotations.destructiveHint = false;
+        } else {
+            annotations.destructiveHint = this.operationType === "delete";
+        }
```

Comment:
```suggestion
        switch (this.operationType) {
        	case "read":
        	case "metadata":
			annotations.readOnlyHint = true;
			annotations.destructiveHint = false;
			break;
	        case "delete": 
			annotations.destructiveHint = this.operationType === "delete";
			break;
	}
```
very nit and subjective just seems easier to understand the branching